src/Compilers/CSharp/Portable/Symbols/PublicModel/TypeSymbol.cs

```diff
@@ -168,5 +168,7 @@ ImmutableArray<SymbolDisplayPart> ITypeSymbol.ToMinimalDisplayParts(SemanticMode
         bool ITypeSymbol.IsRefLikeType => UnderlyingTypeSymbol.IsRefLikeType;
 
         bool ITypeSymbol.IsReadOnly => UnderlyingTypeSymbol.IsReadOnly;
+
+        bool ITypeSymbol.IsRecord => SymbolDisplayVisitor.FindValidCloneMethod(this) != null;
```

Comment:
>SymbolDisplayVisitor.FindValidCloneMethod(this) [](start = 37, length = 47)

I think ```SynthesizedRecordClone.FindValidCloneMethod(UnderlyingTypeSymbol, ref useSiteDiagnostics)``` will be preferred here. It will avoid realizing ISymbol instances for methods within the type.

Reply:
Thank you for the suggestions. I've removed SymbolDisplayVisitor.FindValidCloneMethod and replaced it with SynthesizedRecordClone.FindValidCloneMethod as suggested.